crates/bevy_pbr/src/deferred/pbr_deferred_functions.wgsl

```diff
@@ -0,0 +1,146 @@
+// ---------------------------
+
+
+// Creates the deferred gbuffer from a PbrInput.
+fn deferred_gbuffer_from_pbr_input(in: PbrInput) -> vec4<u32> {
+     // Only monochrome occlusion supported. May not be worth including at all.
+     // Some models have baked occlusion, GLTF only supports monochrome. 
+     // Real time occlusion is applied in the deferred lighting pass.
+    let occlusion = dot(in.occlusion, vec3<f32>(0.2126, 0.7152, 0.0722));
+#ifdef WEBGL // More crunched for webgl so we can also fit depth.
+    var props = deft::pack_unorm3x4_plus_unorm_20_(vec4(
+        in.material.reflectance,
+        in.material.metallic,
+        occlusion, 
+        in.frag_coord.z));
+#else
+    var props = deft::pack_unorm4x8_(vec4(
+        in.material.reflectance, // could be fewer bits
+        in.material.metallic, // could be fewer bits
+        occlusion, // is this worth including?
+        0.0)); // spare
+#endif // WEBGL
+    let flags = deft::deferred_flags_from_mesh_mat_flags(in.flags, in.material.flags);
+    let oct_nor = deft::octa_encode(normalize(in.N));
+    var base_color_srgb = vec3(0.0);
+    var emissive = in.material.emissive.rgb;
+    if ((in.material.flags & STANDARD_MATERIAL_FLAGS_UNLIT_BIT) != 0u) {
+        // Material is unlit, use emissive component of gbuffer for color data.
+        // Unlit materials are effectively emissive.
+        emissive = in.material.base_color.rgb;
+    } else {
+        base_color_srgb = pow(in.material.base_color.rgb, vec3(1.0 / 2.2));
+    }
+    let deferred = vec4(
+        deft::pack_unorm4x8_(vec4(base_color_srgb, in.material.perceptual_roughness)),
+        rgb9e5::vec3_to_rgb9e5_(emissive),
+        props,
+        deft::pack_24bit_nor_and_flags(oct_nor, flags),
+    );
+    return deferred;
+}
+
+// Creates a PbrInput from the deferred gbuffer.
+fn pbr_input_from_deferred_gbuffer(frag_coord: vec4<f32>, gbuffer: vec4<u32>) -> PbrInput {
+    var pbr: PbrInput;
+    pbr.material = standard_material_new();
+
+    let flags = deft::unpack_flags(gbuffer.a);
+    let deferred_flags = deft::mesh_mat_flags_from_deferred_flags(flags);
+    pbr.flags = deferred_flags.x;
+    pbr.material.flags = deferred_flags.y;
+
+    let base_rough = deft::unpack_unorm4x8_(gbuffer.r);
+    pbr.material.perceptual_roughness = base_rough.a;
+    let emissive = rgb9e5::rgb9e5_to_vec3_(gbuffer.g);
+    if ((pbr.material.flags & STANDARD_MATERIAL_FLAGS_UNLIT_BIT) != 0u) {
+        pbr.material.base_color = vec4(emissive, 1.0);
+        pbr.material.emissive = vec4(vec3(0.0), 1.0);
+    } else {
+        pbr.material.base_color = vec4(pow(base_rough.rgb, vec3(2.2)), 1.0);
+        pbr.material.emissive = vec4(emissive, 1.0);
+    }
+#ifdef WEBGL // More crunched for webgl so we can also fit depth.
+    let props = deft::unpack_unorm3x4_plus_unorm_20_(gbuffer.b);
+    // Bias to 0.5 since that's the value for almost all materials.
+    pbr.material.reflectance = saturate(props.r - 0.03333333333); 
```

Comment:
reflectance is nearly always 0.5 for any physically based material. Since that is the most common value we are biasing this to 0.5 since it's getting crunched into just 4 bits for webgl2.

Reply:
4 bits as in mapping to 0, 1/3, 2/3, 1? I still don't understand the explanation, which suggests to me that perhaps others won't too.

Reply:
4 bits as in a 4 bit unorm. So there's 16 representable values.

Reply:
I still don't understand. :/ I'm sure it's well-reasoned I just don't understand why subtracting 0.033333... is the same as biasing the reflectance to 0.5. But I also find the term 'biasing' to be difficult to understand what people mean. Could you try to be super clear about what is going on here?